dist/index.js

```diff
@@ -3932,19 +3940,25 @@ try {
   let docNumber = 1;
   if (yamlDocs.length == 0) {
     failed('No yaml documents detected in ' + yamlFilePath);
+    core.setOutput('validation-outcome', 'failed');
   } else {
     yamlDocs.forEach(doc => {
       info('Validating Document #' + docNumber);
       yamlLibrary.checkDocAgainstSchema(doc, schemaDoc, failed, warn, info);
       if (docFailed) {
         failed('Document #' + docNumber + ' failed validation.');
+      }
+      if (docWarn) {
+        warn('Document #' + docNumber + ' has warnings.');
       } else {
         info('Document #' + docNumber + ' successfully validated.');
       }
       info('Finished Validating Document #' + docNumber);
       docNumber++;
       docFailed = false;
+      docWarn = false;
     });
+    core.setOutput('validation-outcome', hasFailure ? 'failed' : hasWarn ? 'warning' : 'success');
```

Comment:
You can keep it like this but the double ternary is hard to read.  Would it be possible to not do a final core.setOutput and just put a core.setOutput in each of the sections above?

Reply:
The sam.yaml files can have multiple documents. That makes me think there should be an overall result.